### DISCUSSION TITLE
When using useQuery with suspense true option, staleTime defaults to 1000ms right?

### DISCUSSION THREAD

**Discussion by pys99pys** [TanStack/query#5162] [Q&A] [answered]
I looked at the use-query documentation and it says that staleTime defaults to 0.

However, [the code](https://github.com/TanStack/query/blob/main/packages/react-query/src/suspense.ts#L14) seems to set staleTime to 1000 when suspense is true.

Also, when I run the code below, the query seems to remain fresh for 1 second.

```
const { data: users } = useQuery("key", getUsersApi, {
  suspense: true,
});
```

![query](https://user-images.githubusercontent.com/42885446/226190050-35e6dfc8-4d5d-471e-9da0-0680a4df9508.gif)

As expected, when the suspense option is true, the default staleTime becomes 1000ms, right?

Or it's written somewhere in the documentation, but I didn't see it?

**Comment by TkDodo** [comment, marked as answer]
it's an implementation detail, but we cannot default to zero because of how suspense works. suspense unmounts the component and shows the suspense fallback component instead. Then, it re-mounts the component once it's ready. With a staleTime of zero, this would lead to _another_ refetch, which is unwanted. The small staleTime prevents this.

**Reply by pys99pys** [reply]
Thank you for your answer!

**Reply by custardcream98** [reply]
@TkDodo Does setting `refetchOnMount: false` along with `staleTime: 0, gcTime: 0` resolve this issue?
